src/FuncTypeChanging.cpp

```diff
@@ -0,0 +1,131 @@
+#include "FuncTypeChanging.h"
+
+namespace Halide {
+
+static bool operator==(const Var &a, const Var &b) {
+    return a.same_as(b);
+}
+
+namespace FuncTypeChanging {
+
+// NOTE: Precondition: `chunk_idx u< num_chunks`!
+static Expr get_nth_chunk(const Expr &value, const Type &chunk_type,
+                          const Expr &chunk_idx, ChunkOrder chunk_order) {
+    unsigned num_bits_total = value.type().bits();
```

Comment:
(here and elsewhere) `bits` is a signed type (`int`); even though the value is always expected to be positive, convention in Halide is to do the math in matching types